packages/config-plugins/src/ios/Entitlements.ts

```diff
@@ -35,71 +34,83 @@ export function setAssociatedDomains(
   return entitlementsPlist;
 }
 
-export function getEntitlementsPath(projectRoot: string): string {
-  const paths = Paths.getAllEntitlementsPaths(projectRoot);
-  let targetPath: string | null = null;
+export function getEntitlementsPath(
+  projectRoot: string,
+  {
+    targetName,
+    buildConfiguration = 'Release',
+  }: { targetName?: string; buildConfiguration?: string } = {}
+): string | null {
+  const project = getPbxproj(projectRoot);
+  const xcBuildConfiguration = getXCBuildConfigurationFromPbxproj(project, {
+    targetName,
+    buildConfiguration,
+  });
+  if (!xcBuildConfiguration) {
+    return null;
+  }
+  const entitlementsPath = getEntitlementsPathFromBuildConfiguration(
+    projectRoot,
+    xcBuildConfiguration
+  );
+  return entitlementsPath && fs.existsSync(entitlementsPath) ? entitlementsPath : null;
+}
 
-  /**
-   * Add file to pbxproj under CODE_SIGN_ENTITLEMENTS
-   */
+export function getDefaultEntitlementsPath(projectRoot: string) {
   const project = getPbxproj(projectRoot);
   const projectName = getProjectName(projectRoot);
   const productName = getProductName(project);
+  return path.normalize(path.join(projectRoot, 'ios', projectName, `${productName}.entitlements`));
+}
 
-  // Use posix formatted path, even on Windows
-  const entitlementsRelativePath = slash(path.join(projectName, `${productName}.entitlements`));
-  const entitlementsPath = slash(
-    path.normalize(path.join(projectRoot, 'ios', entitlementsRelativePath))
-  );
-
-  const pathsToDelete: string[] = [];
-
-  while (paths.length) {
-    const last = slash(path.normalize(paths.pop()!));
-    if (last !== entitlementsPath) {
-      pathsToDelete.push(last);
-    } else {
-      targetPath = last;
-    }
+function getEntitlementsPathFromBuildConfiguration(
+  projectRoot: string,
+  xcBuildConfiguration: XCBuildConfiguration
+): string | null {
+  const entitlementsPathRaw = xcBuildConfiguration?.buildSettings?.CODE_SIGN_ENTITLEMENTS as
+    | string
+    | undefined;
+  if (entitlementsPathRaw) {
+    return path.normalize(path.join(projectRoot, 'ios', trimQuotes(entitlementsPathRaw)));
+  } else {
+    return null;
   }
+}
 
-  // Create a new entitlements file
-  if (!targetPath) {
-    targetPath = entitlementsPath;
-
-    // Use the default template
-    let template = ENTITLEMENTS_TEMPLATE;
+export function ensureApplicationTargetEntitlementsFileConfigured(projectRoot: string): void {
+  const project = getPbxproj(projectRoot);
+  const projectName = getProjectName(projectRoot);
+  const productName = getProductName(project);
```

Comment:
It would be nice if we exposed a way to pass in the `productName` and `projectName` optionally. In something like App Clips you may want to use this method to ensure the clip entitlements are configured.

Reply:
I added this method just to keep legacy behavior, my preference would be to remove it entirely and handle that in the write method of entitlements mod, unfortunately, it's not possible to modify xcodeproj there because changes would be overridden.

Also, this method is not needed for anything for targets different than main application target
- if you are not adding any entitlements then nothing needs to be added
- if you want to add entitlements to some target, you are already adding that target, so specifying CODE_SIGN_ENTITLEMENTS there is easy